railties/lib/rails/application/configuration.rb

```diff
@@ -120,6 +120,10 @@ def load_defaults(target_version)
           if respond_to?(:action_view)
             action_view.default_enforce_utf8 = false
           end
+
+          if respond_to?(:action_controller) && !Rails.env.production?
```

Comment:
Since the documentation state that:

> The default value is `:raise` in development and test environments, and `false` in all other environments.

This conditional actually cause the error to be raise if user is using another closer-to-production environment, such as staging. I think it's better to be explicit on which environment we want to raise:

```ruby
if respond_to?(:action_controller) && %w(development test).include?(Rails.env)
```